streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -257,9 +257,11 @@ void handleRevocation(final Collection<TopicPartition> revokedPartitions) {
         final Set<TopicPartition> remainingPartitions = new HashSet<>(revokedPartitions);
 
         for (final Task task : tasks.values()) {
-            if (remainingPartitions.containsAll(task.inputPartitions())) {
-                revokedTasks.add(task.id());
-                remainingPartitions.removeAll(task.inputPartitions());
+            for (final TopicPartition topicPartition : task.inputPartitions()) {
```

Comment:
Can you elaborate a bit on this fix? Do you mean one task is subscribed to a pattern that matches multiple topics, or to two different patterns? I'm not sure I understand why either case would cause `containsAll` to fail.

Reply:
In fact, I think this fix is not necessary with the other fix. Could be removed.

Reply:
Hmm.. why it's possible that only part of the partitions of a task is revoked? We do assignment at the granularity of tasks so this check is to verify specifically that all partitions should be included if a task is going to be removed right?

Reply:
I think now I understand the reason that part of the partitions can be removed here. 

If a task is indeed being removed, it should be triggered in the TaskManager#onAssignment; here TaskManager#onRevocation is triggered after the previous call so the tasks map should have been updated --- i.e. the task would not be inside task-manager anymore, and if it is due to regex pattern (like this test) the current condition is okay: we should not suspend the task unless all its input-partitions are included. Otherwise, we can just update the input partition of that task --- right now it is final so we cannot update it, but I think that’s fine since we would no longer pipe any records to that task and there will be no committed offsets for that partition either --- in either case, we can remove it from the remainingPartitions.